rust/garbage_collector/src/config.rs

```diff
@@ -34,7 +38,11 @@ pub(super) struct GarbageCollectorConfig {
     pub(super) version_cutoff_time: Duration,
     pub(super) max_collections_to_gc: u32,
     pub(super) gc_interval_mins: u32,
-    pub(super) disallow_collections: Vec<String>,
+    #[serde(default = "GarbageCollectorConfig::default_min_versions_to_keep")]
+    pub(super) min_versions_to_keep: u32,
+    #[serde(default = "GarbageCollectorConfig::default_filter_min_versions_if_alive")]
+    pub(super) filter_min_versions_if_alive: Option<u64>,
```

Comment:
these are two separate config fields when they could be one (`filter_min_versions_if_alive` can be `min_versions_to_keep + 1`)
I made them separate fields because otherwise it is difficult to test GC behavior when a collection with no eligible versions is processed